Cargo.lock

```diff
@@ -1193,12 +1177,10 @@ dependencies = [
 [[package]]
 name = "circuit_definitions"
 version = "0.1.0"
-source = "git+https://github.com/matter-labs/era-zkevm_test_harness.git?branch=v1.4.0#1934433af48520ac70dc3080b0d1feba877a07c5"
+source = "git+https://github.com/matter-labs/era-zkevm_test_harness.git?branch=v1.4.0#43aeb53d7d9c909508a98f9fc140edff0e9d2357"
```

Comment:
Changes like this are extremely dangerous. There is no notion of semver and stuff in many crypto repos, so simply changing a git revision may be a breaking change requiring an update.

I would be extremely scared if I had to merge this PR, chances of it breaking everything are very high.

Reply:
So what's the right approach of upgrading stuff around here?

Reply:
Well, I think you won't like the answer. We normally update deps 1 by 1 when strictly necessary 😓

Reply:
(...and sometimes still break stuff)

Reply:
I've consulted @akash-chandrakar . I've manually bumped the minimal set of packages and run the prover integration test against it.